### ISSUE TITLE
Rate limiter and history should be managed in the database, not in memory

### ISSUE THREAD

**Issue by kenellorando** [kenellorando/cadence#230]
Since the rate limiter and history manage data in-memory, they do not function properly when Cadence scales beyond one instance, or when the lone instance of Cadence needs to be redeployed. 

If we moved this data into a separate database, we would be properly separate the API from this data, and only keep one master copy of it instead of having each replica duplicate it.

Depends on #224.

**Comment by kenellorando** [comment]
So far this is half done: rate limiter is in database but song history is still in-memory.

I'm going to edit this issue to relate to the rate limiter only, then split off history into a new issue (it's a little bit trickier).
